src/amd/compiler/aco_instruction_selection.cpp

```diff
@@ -4207,7 +4207,7 @@ void visit_phi(isel_context *ctx, nir_phi_instr *instr)
    aco_ptr<Instruction> phi;
    unsigned num_src = exec_list_length(&instr->srcs);
    Temp dst = get_ssa_temp(ctx, &instr->dest.ssa);
-   aco_opcode opcode = ctx->divergent_vals[instr->dest.ssa.index] ? aco_opcode::p_phi : aco_opcode::p_linear_phi;
+   aco_opcode opcode = dst.is_linear() ? aco_opcode::p_linear_phi : aco_opcode::p_phi;
```

Comment:
iirc, there was a reason for that.
did you hit a bug with that?
the linear_phi indicates that the register moves have to be placed on the incoming linear/physical edges.